### PR TITLE
DAH-1869 fix: use dayjs to handle various startTime formats

### PR DESCRIPTION
**Pull request by cade-exygy**
Jira: https://sfgovdt.jira.com/browse/DAH-1869

# Changes
- use dayjs to convert date string to Dayjs object
- use dayjs.diff to sort dates

# Review
-Go to a Sales listing: /listings/a0W4U00000KneiUUAR?react=true
-Ensure that the open houses are correctly ordered by date and start time
-You can compare with the Angular version: /listings/a0W4U00000KneiUUAR
-You can edit the open house list in Salesforce as well to test

### CODE REVIEW

File: app/javascript/util/listingUtil.ts [hunk 1]
```diff
@@ -181,11 +181,13 @@ const formatEventTime = (eventTime: string) => {
 
 export const getEventDateTime = (eventDate: string, eventTime: string) => {
   const startTime = eventTime?.includes(":") ? eventTime : `${formatEventTime(eventTime)}`
-  return new Date(`${eventDate} ${startTime}`)
+  const customParseFormat = require("dayjs/plugin/customParseFormat")
```
Comment (jimlin-sfgov) [inline app/javascript/util/listingUtil.ts:184]: we should stick to `import` for consistency

**Review by jimlin-sfgov — CHANGES_REQUESTED** [review]

**Comment by jimlin-sfgov** [comment]
In the future, let's hold off on merging approved PRs until they pass PA testing. That way we avoid multiple PRs per ticket, like this one.

**Comment by cade-exygy** [comment]
> In the future, let's hold off on merging approved PRs until they pass PA testing. That way we avoid multiple PRs per ticket, like this one.

Fair point. We may have to adjust how we do PA. I believe PA testing is primarily done on the full instance. Also, we will have to make some tweaks to ensure that the PR deployment environment has the same vars and plugins as Full.

**Comment by jimlin-sfgov** [comment]
> I believe PA testing is primarily done on the full instance

Yeah, we will have to do some tweaks before fully moving to PR instances for PA testing. For PRs that do not rely on environment vars etc., I have been commenting in the Jira ticket with the PR instance URL for PA testers to use.

**Review by jimlin-sfgov — APPROVED** [review]
Looks good.

The November ones are out of order, but that is because of inconsistent user input in salesforce.

The custom parsing format here handles what salesforce users _usually_ use, which is the best we can do on the front end.

**Comment by cade-exygy** [comment]
> Looks good.
> 
> The November ones are out of order, but that is because of inconsistent user input in salesforce.
> 
> The custom parsing format here handles what salesforce users _usually_ use, which is the best we can do on the front end.

Awesome! Moved ticket to PA.

**Pull request merged**